komodo/release_transpiler.py

```diff
@@ -0,0 +1,130 @@
+            "py36" : "3.6.9"
+        },
+        "rhel7": {
+            "py27" : "2.7.14+builtin",
+            "py36" : "3.6.9+builtin"
+        }
+    },
+    "qt5" : {
+        "rhel6": {
+            "py27" : None,
+            "py36" : "5.11.13",
+        },
+        "rhel7": {
+            "py27" : None,
+            "py36" : "5.11.13+builtin",
+        }
+    }
+}
+
+
+def build_matrix_file(release_base, release_folder, builtins):
+    py27 = load_yaml("{}/{}-py27.yml".format(release_folder, release_base))
+
+    py36 = load_yaml("{}/{}-py36.yml".format(release_folder, release_base))
+
+    all_packages = set(py36.keys()).union(py27.keys())
+
+    compiled = {}
+
+    for p in all_packages:
+        if p in builtins:
+            compiled[p] = builtins[p]
+        elif py27.get(p) == py36.get(p):
+            compiled[p] = py27.get(p)
+        else:
+            compiled[p] = {"py27": py27.get(p), "py36": py36.get(p)}
+
+    write_to_file(compiled, "{}.yml".format(release_base), False)
+
+def _build(packages, py_ver, rhel_ver):
+    release_dict = {}
+    for p, versions in packages.items():
+        if rhel_ver in versions:
+            version = versions[rhel_ver][py_ver]
+        elif py_ver in versions:
+            version = versions[py_ver]
+        else:
+            version = versions
+
+        if version:
+            release_dict[p] = version
+    return release_dict
+
+
+def transpile_releases(release_base, release_folder, output_folder):
+    release_matrix = load_yaml("{}.yml".format(os.path.join(release_folder, release_base)))
+    for rhel_ver in ("rhel6", "rhel7"):
+        for py_ver in ("py27", "py36"):
+            release_dict = _build(release_matrix, py_ver, rhel_ver)
+            filename = "{rel}-{pyver}-{rhel_version}.yml".format(
+                rel=release_base, pyver=py_ver, rhel_version=rhel_ver
+            )
+            write_to_file(release_dict, os.path.join(output_folder, filename))
+
+
+def combine(args):
+    build_matrix_file(args.release_base, args.release_folder, rhel7_builtins)
+
+def transpile(args):
+    transpile_releases(args.release_base, args.release_folder, args.output_folder)
+
+
+
+def main():
+    parser = ArgumentParser(description="Build release files.")
+
+    subparsers = parser.add_subparsers(
+        title="Available user entries",
+        description="time to tidy up",
+        help="Available sub commands",
+        dest="mode",
+    )
+
+    matrix_parser = subparsers.add_parser(
+        "combine"
+    )
+    matrix_parser.set_defaults(func=combine)
+    matrix_parser.add_argument(
+        "--release-base",
+        help="Name of the release to handle"
+    )
+    matrix_parser.add_argument(
+        "--release-folder",
+        help="Folder with existing release file"
+    )
+
+    transpile_parser = subparsers.add_parser(
+        "transpile"
+    )
+    transpile_parser.set_defaults(func=transpile)
+    transpile_parser.add_argument(
+        "--release-base",
```

Comment:
Why not just accept the matrix file directly? `matrix-file` 

And not an optional argument.